cypress/integration/regression/main-table-regression.spec.ts

```diff
@@ -0,0 +1,226 @@
+    mainTablePage.mainTable.searchColumn.clear()
+    mainTablePage.mainTable.searchColumn.type('Gene')
+    mainTablePage.mainTable.columnSwitch.eq(1).click({ force: true })
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.columnHeader.countElements(8)
+  })
+
+  it('should turn off column from the middle part of a table | step 9', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: Timeouts.TwentySecondsTimeout })
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.customizeTable.click()
+    mainTablePage.mainTable.searchColumn.type('In-Silico')
+    mainTablePage.mainTable.columnSwitch.eq(1).click({ force: true })
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.columnHeader.countElements(7)
+  })
+
+  it('should turn on column from the middle part of table | step 10', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: Timeouts.TwentySecondsTimeout })
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.customizeTable.click()
+    mainTablePage.mainTable.searchColumn.type('In-Silico')
+    mainTablePage.mainTable.customizeTableList.element.within($list => {
+      $list.find(CommonSelectors.columnSwitch).trigger('click')
+    })
+    //mainTablePage.mainTable.columnSwitch.eq(1).click()
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.columnHeader.countElements(7)
+    mainTablePage.mainTable.customizeTable.click()
+    mainTablePage.mainTable.searchColumn.clear()
+    mainTablePage.mainTable.searchColumn.type('In-Silico')
+    mainTablePage.mainTable.columnSwitch.eq(1).click({ force: true })
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.columnHeader.countElements(8)
+  })
+
+  it('should save Excel file | test #12', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: Timeouts.TwentySecondsTimeout })
+    mainTablePage.mainTable.selectPreset.click()
+    cy.intercept('POST', '/app/tab_report').as('loadPreset')
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.preset.getButtonByText(homozygous).click()
+    cy.wait('@loadPreset', { timeout: Timeouts.TwentySecondsTimeout })
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+    cy.intercept('GET', '/app/excel/PGP3140_wgs_panel_hl_****.xlsx').as(
+      'reportDownload',
+    )
+    mainTablePage.mainTable.exportReport.click()
+    mainTablePage.mainTable.exportExcel.click()
+    cy.wait('@reportDownload', { timeout: Timeouts.FifteenSecondsTimeout })
+    cy.readFile('./cypress/downloads/PGP3140_wgs_panel_hl.xlsx').should('exist')
```

Comment:
I'd suggest to clear downloads directory before the run to be sure the new file was downloaded. Via npm scripts.

Reply:
made it using TrashAssetsBeforeRun = true